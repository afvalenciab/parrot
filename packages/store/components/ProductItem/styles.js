import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles({
  card: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  wrapperContent: {
    display: 'flex',
  },
  cardMedia: {
    borderRadius: '.5rem',
    margin: '.5rem',
    overflow: 'hidden',
    flexShrink: 0,
  },
  cardContent: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    paddingBottom: '0 !important',
    padding: '0 0 0 .5rem',
  },
  productName: {
    fontWeight: 'bold',
    fontSize: '1.1rem',
  },
  cardActions: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignItems: 'end',
    paddingRight: '1rem',
  },
  available: {
    color: 'green',
  },
  unavailable: {
    color: 'red',
  },
});
