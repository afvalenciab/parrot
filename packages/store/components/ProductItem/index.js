import { Card, CardMedia, CardContent, CardActions, Switch, Typography } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import Image from 'next/image';
import clsx from 'clsx';

import { useDispatcherStores } from 'providers/Stores/useDispatcher';

import { useStyles } from './styles';

export default function ProductItem({ product }) {
  const theme = useTheme();
  const classes = useStyles(theme);
  const { editProduct } = useDispatcherStores();

  const status = {
    isAvailable: product?.availability === 'AVAILABLE',
    label: product?.availability === 'AVAILABLE' ? 'Disponible' : 'No disponible',
  };

  const handleChange = ({ target }) => {
    editProduct({ ...product, availability: target.checked ? 'AVAILABLE' : 'UNAVAILABLE' });
  };

  return (
    <Card className={classes.card} data-testid={`card-${product.uuid}`}>
      <div className={classes.wrapperContent}>
        <CardMedia className={classes.cardMedia}>
          <Image src={product?.imageUrl} width={100} height={100} alt={product?.name} priority />
        </CardMedia>

        <CardContent className={classes.cardContent}>
          <Typography variant="subtitle2" component="p" className={classes.productName}>
            {product?.name}
          </Typography>

          <Typography variant="subtitle2" component="p">
            {`$${product?.price} MXN`}
          </Typography>
        </CardContent>
      </div>

      <CardActions className={classes.cardActions}>
        <Switch checked={status.isAvailable} size="medium" onChange={handleChange} />

        <Typography
          variant="caption"
          component="p"
          className={clsx({
            [classes.available]: status.isAvailable,
            [classes.unavailable]: !status.isAvailable,
          })}>
          {status.label}
        </Typography>
      </CardActions>
    </Card>
  );
}
