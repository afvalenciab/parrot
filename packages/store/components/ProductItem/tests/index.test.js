import React from 'react';
import { render } from '@testing-library/react';

import useDispatcherStores from 'providers/Stores/useDispatcher';
import ProductItem from 'components/ProductItem';

jest.mock('providers/Stores/useDispatcher');

describe('<ProductItem />', () => {
  const props = {
    product: {
      uuid: '123',
      availability: 'AVAILABLE',
      imageUrl: 'https://iamge.com',
      name: 'Test 1',
      price: '200',
    },
  };

  const defaultUseDispatcherStores = { editProduct: jest.fn() };

  beforeEach(() => {
    useDispatcherStores.mockReturnValue(defaultUseDispatcherStores);
  });

  it('Should render correctly', () => {
    const { container } = render(<ProductItem {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render the card-123 when the component is mounted', () => {
    const { getByTestId } = render(<ProductItem {...props} />);

    const card = getByTestId('card-123');

    expect(card).toBeInTheDocument();
  });
});
