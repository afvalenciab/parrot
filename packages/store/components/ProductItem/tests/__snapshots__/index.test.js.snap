// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem /> Should render correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="card-123"
  >
    <div
      class="makeStyles-wrapperContent-2"
    >
      <div
        class="MuiCardMedia-root makeStyles-cardMedia-3"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjEwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Test 1"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fiamge.com&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fiamge.com&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fiamge.com&w=256&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="MuiCardContent-root makeStyles-cardContent-4"
      >
        <p
          class="MuiTypography-root makeStyles-productName-5 MuiTypography-subtitle2"
        >
          Test 1
        </p>
        <p
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          $200 MXN
        </p>
      </div>
    </div>
    <div
      class="MuiCardActions-root makeStyles-cardActions-6 MuiCardActions-spacing"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-10 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-12 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <p
        class="MuiTypography-root makeStyles-available-7 MuiTypography-caption"
      >
        Disponible
      </p>
    </div>
  </div>
</div>
`;
